storagenode/piecestore/endpoint.go

```diff
@@ -352,6 +360,11 @@ func (endpoint *Endpoint) doUpload(stream uploadStream) (err error) {
 				return ErrInternal.Wrap(err)
 			}
 
+			// Save the order before completing the call. Set orderSaved so
+			// that the defer above does not also save.
+			orderSaved = true
+			endpoint.saveOrder(ctx, limit, &largestOrder)
```

Comment:
The reason it's not doing it before sending the final hash is not to block the uplink for that amount of time.

Reply:
Fair enough. I'll revert and block on the request count dropping to zero.

Reply:
I think that we should revive this commit, because we have found out that we actually _are_ blocking the uplink during this amount of time (when using grpc), and it prevents accidental race conditions. Thoughts?